File: frontend/my-app/src/lib/presupuesto-observaciones.ts
const PREFIJO_CONJUNTOS_ARMADOS = "Conjuntos ya armados para esta fecha:";

/** Texto interno de conjuntos armados que no debe guardarse ni enviarse al cliente. */
export function esObservacionConjuntosArmados(
  observaciones?: string | null
): boolean {
  if (!observaciones?.trim()) return false;
  return observaciones.trimStart().startsWith(PREFIJO_CONJUNTOS_ARMADOS);
}

/** Observaciones reales para persistir (excluye aviso interno de conjuntos). */
export function observacionesParaGuardar(
  observaciones?: string | null
): string {
  if (!observaciones?.trim() || esObservacionConjuntosArmados(observaciones)) {
    return "";
  }
  return observaciones.trim();
}

/** Observaciones para mensajes al cliente (WhatsApp, etc.). */
export function observacionesParaCliente(
  observaciones?: string | null
): string | null {
  const limpio = observacionesParaGuardar(observaciones);
  return limpio || null;
}

File: backend/src/presupuesto_observaciones.py
"""Observaciones de presupuesto visibles para cliente/orden (excluye avisos internos)."""

PREFIJO_CONJUNTOS_ARMADOS = "Conjuntos ya armados para esta fecha:"


def es_observacion_conjuntos_armados(observaciones: str | None) -> bool:
    if not (observaciones or "").strip():
        return False
    return observaciones.strip().startswith(PREFIJO_CONJUNTOS_ARMADOS)


def observaciones_presupuesto_para_mostrar(observaciones: str | None) -> str:
    """Texto de observaciones útil para orden, reportes y modista (sin aviso interno)."""
    if not (observaciones or "").strip():
        return ""
    if es_observacion_conjuntos_armados(observaciones):
        return ""
    return observaciones.strip()
